tests/conftest.py

```diff
@@ -19,6 +29,167 @@
+
+        yield engine
+
+
+@pytest.fixture(scope="function")
+def benchmark_db_session(benchmark_db_engine):
+    """SQLAlchemy session for a given test.
+
+    Most tests shouldn't need to include this fixture directly.
+
+    Yields
+    ------
+    SQLAlchemy session for the benchmark database engine if run as part of a benchmark,
+    None otherwise.
+    """
+    if not benchmark_db_engine:
+        yield
+    else:
+        with Session(benchmark_db_engine, future=True) as session, session.begin():
+            yield session
+
+
+@pytest.fixture(scope="function")
+def test_run_benchmark(benchmark_db_session, request, testrun_uid):
+    """SQLAlchemy ORM object representing a given test run.
+
+    By including this fixture in a test (or another fixture that includes it)
+    you trigger the test being written to the benchmark database. This fixture
+    includes data common to all tests, including python version, test name,
+    and the test outcome.
+
+    Yields
+    ------
+    SQLAlchemy ORM object representing a given test run if run as part of a benchmark,
+    None otherwise.
+    """
+    if not benchmark_db_session:
+        yield
+    else:
+        run = TestRun(
+            session_id=testrun_uid,
+            name=request.node.name,
+            originalname=request.node.originalname,
+            path=str(request.node.path.relative_to(TEST_DIR)),
+            cubed_version=cubed.__version__,
+            #cubed_xarray_version=cubed_xarray.__version__,
+            #xarray_version=xarray.__version__,
+            #lithops_version=lithops.__version__, 
+            python_version=".".join(map(str, sys.version_info)),
+            platform=sys.platform,
+            ci_run_url=WORKFLOW_URL,
+        )
+        yield run
+
+        rep = getattr(request.node, "rep_setup", None)
+        if rep:
+            run.setup_outcome = rep.outcome
+        rep = getattr(request.node, "rep_call", None)
+        if rep:
+            run.call_outcome = rep.outcome
+        rep = getattr(request.node, "rep_teardown", None)
+        if rep:
+            run.teardown_outcome = rep.outcome
+
+        benchmark_db_session.add(run)
+        benchmark_db_session.commit()
+
+
+@pytest.fixture(scope="function")
+def benchmark_time(test_run_benchmark):
+    """
+    Benchmark the wall clock time of executing some code.
+
+    Yields
+    ------
+    Context manager that records the wall clock time duration of executing
+    the ``with`` statement if run as part of a benchmark, or does nothing otherwise.
+
+    Example
+    -------
+    .. code-block:: python
+
+        def test_something(benchmark_time):
+            with benchmark_time:
+                do_something()
+    """
+
+    @contextlib.contextmanager
+    def _benchmark_time():
+        if not test_run_benchmark:
+            yield
+        else:
+            start = time.time()
+            yield
+            end = time.time()
```

Comment:
At the moment I think this will time the execution of the entire test, but I think we want to change that to be able to ignore setting up (e.g. by placing some random initial data in a bucket).

Reply:
This now just times whatever happens once you call `cubed.compute`. But that may or may not exclude the creation of the initial data in the `quad_means` test, I'm not sure.